pennylane_cirq/cirq_device.py

```diff
@@ -69,23 +78,31 @@ class CirqDevice(QubitDevice, abc.ABC):
     short_name = "cirq.base_device"
 
     def __init__(self, wires, shots, analytic, qubits=None):
-        super().__init__(wires, shots, analytic)
-
-        self.circuit = None
 
-        device_wires = self.map_wires(self.wires)
+        if not isinstance(wires, Iterable):
+            # interpret wires as the number of consecutive wires
+            wires = range(wires)
+        num_wires = len(wires)
 
         if qubits:
-            if wires != len(qubits):
+            if num_wires != len(qubits):
                 raise qml.DeviceError(
                     "The number of given qubits and the specified number of wires have to match. Got {} wires and {} qubits.".format(
                         wires, len(qubits)
                     )
                 )
-
-            self.qubits = qubits
         else:
-            self.qubits = [cirq.LineQubit(wire) for wire in device_wires.labels]
+            qubits = [cirq.LineQubit(idx) for idx in range(num_wires)]
+
+        # cirq orders the subsystems based on a total order defined on qubits.
+        # For consistency, this plugin uses that same total order
+        self._unsorted_qubits = qubits
+        self.qubits = sorted(qubits)
+
+        super().__init__(wires, shots, analytic)
```

Comment:
I don't understand why `super` is not called earlier? That would remove a lot of code above, because `num_wires` can be extracted from `len(self.wires)` or `self.num_wires`. One could still raise an error if ``qubits`` has a different length.

Reply:
@mariaschuld, here is @co9olguy's original comment: https://github.com/PennyLaneAI/pennylane-cirq/pull/40#discussion_r471025961

> I had to move down the superclass initialization here, since it is the thing that creates the wire_map, but the wire_map can't be determined until the cirq qubits have been assigned

Reply:
Ah maybe that is due to other changes made in the original PR? In that case we can merge as is, to not to create conflicts!

Reply:
`define_wire_map` is called in the parent class's `__init__` method. 

We don't need to know the number of qubits (as you say, that can be inferred), but we do need to do all this qubit pre-processing before `define_wire_map` makes sense to be called